server.go

```diff
@@ -52,10 +52,23 @@ func (srv *server) shutdownHTTPServer() error {
 // Run implements the Server interface
 func (srv *server) Run() error {
 	// Launch HTTP server
-	if err := srv.httpServer.Serve(
-		tcpKeepAliveListener{srv.listener.(*net.TCPListener)},
-	); err != http.ErrServerClosed {
-		return fmt.Errorf("HTTP Server failure: %s", err)
+	if srv.options.TLS == Enabled {
```

Comment:
This should move to a [new server constructor function](https://github.com/qbeon/webwire-go/pull/34#issuecomment-430401259) which must be similar to [NewServer](https://github.com/qbeon/webwire-go/blob/master/newServer.go#L12) but be implemented with a TLS TCP/IP listener.